### PR TITLE
feat: getting started

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Describe your changes

Fixes https://linear.app/nango/issue/NAN-2122/implement-getting-started-tab 
Fixes https://linear.app/nango/issue/NAN-2123/track-events-video-started-and-ended

- Replace Interactive Demo by Getting Started
I dropped all the codebase related to the interactive demo

- Add missing unique index for `user_id` 
- Add new endpoint `PATCH /onboarding` 
Only used to set onboarding as done

Screenshot 2024-11-19 at 12 04 38

### CODE REVIEW

**Comment by TBonnin** [comment]
First I see `Files changed 49` and I am like 😓 then I see `+459 −1,983` and I am like 🥳

File: packages/database/lib/migrations/20241119110518_onboarding_unique_key.cjs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    await knex.schema.raw(`CREATE UNIQUE INDEX "_nango_onboarding_demo_progress_user_id_index" ON "_nango_onboarding_demo_progress" USING BTREE ("user_id")`);
+};
+
+/**
+ * @param {import('knex').Knex} knex
+ */
+exports.down = async function (knex) {
+    await knex.schema.raw(`DROP INDEX "_nango_onboarding_demo_progress_user_id_index"`);
+    await knex.schema.raw(`CREATE INDEX "_nango_onboarding_demo_progress_user_id_index" ON "_nango_onboarding_demo_progress" USING BTREE ("user_id")`);
+};
```
Comment (TBonnin) [inline packages/database/lib/migrations/20241119110518_onboarding_unique_key.cjs:17]: I am not even sure why we bother with `down` queries to be honest

File: packages/webapp/src/pages/GettingStarted/Show.tsx [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    }, []);
+
+    useScript('https://www.youtube.com/iframe_api');
+
+    const triggerVideo = () => {
+        if (hasVideo) {
+            return;
+        }
+        if (!ytLoaded) {
+            // adblock
```
Comment (TBonnin) [inline packages/webapp/src/pages/GettingStarted/Show.tsx:36]: adblock?

**Review by TBonnin — APPROVED** [review]

File: packages/database/lib/migrations/20241119110518_onboarding_unique_key.cjs (hunk 1 above)
Comment (bodinsamuel) [reply packages/database/lib/migrations/20241119110518_onboarding_unique_key.cjs:17]: you are preaching to the converted here :D

File: packages/webapp/src/pages/GettingStarted/Show.tsx (hunk 2 above)
Comment (bodinsamuel) [reply packages/webapp/src/pages/GettingStarted/Show.tsx:36]: If YouTube does not load it's most of the time adblocked (third party script blocked), I didn't really bothered finding a solution. Most of the time people add youtube the allowlist anyway

**Pull request merged**